crates/polars-parquet/src/parquet/encoding/hybrid_rle/encoder.rs

```diff
@@ -70,31 +169,125 @@ fn bitpacked_encode_u32<W: Write, I: Iterator<Item = u32>>(
             .for_each(|(item, buf)| *buf = item);
 
         let mut packed = [0u8; 4 * U32_BLOCK_LEN];
-        bitpacked::encode_pack(&buffer, num_bits, packed.as_mut());
+        bitpacked::encode_pack(&buffer[..remainder], num_bits, packed.as_mut());
         writer.write_all(&packed[..compressed_remainder_size])?;
     };
     Ok(())
 }
 
-/// the bitpacked part of the encoder.
+fn run_length_encode_u32<W: Write>(
+    writer: &mut W,
+    run_length: usize,
+    value: u32,
+    bit_width: u32,
+) -> std::io::Result<()> {
+    // write the length + indicator
+    let mut header = run_length as u64;
+    header <<= 1;
+    let mut container = [0; 10];
+    let used = uleb128::encode(header, &mut container);
+    writer.write_all(&container[..used])?;
+
+    let num_bytes = ceil8(bit_width as usize);
+    let bytes = value.to_le_bytes();
+    writer.write_all(&bytes[..num_bytes])?;
+    Ok(())
+}
+
+#[allow(clippy::comparison_chain)]
 pub fn encode_bool<W: Write, I: Iterator<Item = bool>>(
```

Comment:
This is the same logic as `encode_u32`, can we make a single generic and dispatch this function to that.

Reply:
I'm very new to Rust and took a while to figure this out. Not sure if I implemented this correctly.